include/sycl_ext_complex.hpp

```diff
@@ -1518,10 +1518,559 @@ operator<<(const sycl::stream &__ss, const complex<_Tp> &_x) {
+      rtn[i] = lhs[i] op rhs;                                                  \
+                                                                               \
+    return rtn;                                                                \
+  }                                                                            \
+                                                                               \
+  friend marray<bool, NumElements> operator op(const DataT &lhs,               \
+                                               const marray &rhs) {            \
+    marray<bool, NumElements> rtn;                                             \
+    for (std::size_t i = 0; i < NumElements; ++i)                              \
+      rtn[i] = lhs op rhs[i];                                                  \
+                                                                               \
+    return rtn;                                                                \
+  }
+
+  OP(==)
+  OP(!=)
+
+#undef OP
+
+  // OP is: <, >, <=, >=
+#define OP(op)                                                                 \
+  friend marray<bool, NumElements> operator op(const marray &lhs,              \
+                                               const marray &rhs) = delete;    \
+  friend marray<bool, NumElements> operator op(const marray &lhs,              \
+                                               const DataT &rhs) = delete;     \
+  friend marray<bool, NumElements> operator op(const DataT &lhs,               \
+                                               const marray &rhs) = delete;
+
+  OP(<);
+  OP(>);
+  OP(<=);
+  OP(>=);
+
+#undef OP
+
+  friend marray operator~(const marray &v) = delete;
+
+  friend marray<bool, NumElements> operator!(const marray &v) = delete;
+};
+
+_SYCL_END_NAMESPACE
+
+_SYCL_EXT_CPLX_BEGIN_NAMESPACE
+
+// Make_complex_marray
+
+template <class T, std::size_t NumElements>
+sycl::marray<complex<T>, NumElements>
+make_complex_marray(const sycl::marray<T, NumElements> &real,
+                    const sycl::marray<T, NumElements> &imag) {
+  sycl::marray<complex<T>, NumElements> rtn;
+
+  for (std::size_t i = 0; i < NumElements; ++i) {
+    rtn[i].real(real[i]);
+    rtn[i].imag(imag[i]);
+  }
+
+  return rtn;
+}
+
+template <class T, std::size_t NumElements>
+sycl::marray<complex<T>, NumElements>
+make_complex_marray(const sycl::marray<T, NumElements> &real, const T &imag) {
+  sycl::marray<complex<T>, NumElements> rtn;
+
+  for (std::size_t i = 0; i < NumElements; ++i) {
+    rtn[i].real(real[i]);
+    rtn[i].imag(imag);
+  }
+
+  return rtn;
+}
+
+template <class T, std::size_t NumElements>
+sycl::marray<complex<T>, NumElements>
+make_complex_marray(const T &real, const sycl::marray<T, NumElements> &imag) {
+  sycl::marray<complex<T>, NumElements> rtn;
+
+  for (std::size_t i = 0; i < NumElements; ++i) {
+    rtn[i].real(real);
+    rtn[i].imag(imag[i]);
+  }
+
```

Comment:
Do we really want those? They sound super ugly...

Reply:
Thanks! I would like to keep this namespace as small as possible.  Extending CATCH2 Macro sound like a good path :)